extensions/ql-vscode/src/queries-panel/queries-module.ts

```diff
@@ -6,12 +6,42 @@ import { DisposableObject } from "../common/disposable-object";
 import { QueriesPanel } from "./queries-panel";
 import { QueryDiscovery } from "./query-discovery";
 import { QueryPackDiscovery } from "./query-pack-discovery";
+import { QueriesPanelCommands } from "../common/commands";
 
 export class QueriesModule extends DisposableObject {
+  private queriesPanel: QueriesPanel | undefined;
+
   private constructor(readonly app: App) {
     super();
   }
 
+  public static initialize(
+    app: App,
+    cliServer: CodeQLCliServer,
+  ): QueriesModule {
+    const queriesModule = new QueriesModule(app);
+    app.subscriptions.push(queriesModule);
+
+    queriesModule.initialize(app, cliServer);
+    return queriesModule;
+  }
+
+  public getCommands(): QueriesPanelCommands {
+    if (!isCanary() || !showQueriesPanel()) {
```

Comment:
If we're going to use this check in multiple places, can we pull it out to a function so we don't duplicate the logic or the comment. Maybe into a `shouldInitializeQueriesPanel` method.

Sadly the comment is slightly outdated already because of the merge of https://github.com/github/vscode-codeql/pull/2528, so that's a perfect example of why it's good to avoid duplication.

Reply:
👍 I extracted the check into a method. We can discuss where to push the guard in your other comment.